hs_core/views/resource_folder_hierarchy.py

```diff
@@ -83,16 +82,22 @@ def data_store_structure(request):
             logical_file_type = ''
             logical_file_id = ''
             for f in ResourceFile.objects.filter(object_id=resource.id):
-                if name_with_rel_path == get_resource_file_name_and_extension(f)[0]:
+                if name_with_full_path == f.storage_path:
                     f_pk = f.pk
                     f_url = get_resource_file_url(f)
                     if resource.resource_type == "CompositeResource":
                         logical_file_type = f.logical_file_type_name
                         logical_file_id = f.logical_file.id
                     break
 
-            files.append({'name': fname, 'size': size, 'type': mtype, 'pk': f_pk, 'url': f_url,
-                          'logical_type': logical_file_type, 'logical_file_id': logical_file_id})
+            if f_pk != '':
+                files.append({'name': fname, 'size': size, 'type': mtype, 'pk': f_pk, 'url': f_url,
+                              'logical_type': logical_file_type,
+                              'logical_file_id': logical_file_id})
+            else:
+                logger.error("data_store_structure: filename {} in iRODs has no analogue in Django"
+                             .format(name_with_full_path))
+            # TODO: flag when files in Django have no analogue in iRODs

```

Comment:
@alvacouch I get the other way requires a comprehensive scan the way that ```check_irods_files``` does, hence raised the question in the first place. But I don't get the other ```because``` part regarding the reporting not slowing down code and no extra code beyond accomplishing the task used in inferring it. Do you mean to have a celery task doing this inferring check? We have ```check_irods_files``` management command as well as other debug commands you wrote that can be fired up as needed when we observed issues, so I think we should not complicate the production code part unless it contributes to functionality since any kind of reporting and checking will slow down things in my opinion. Most likely I don't quite get what you have in mind, but since it is a to-do comment, I am not going to worry about it for now.